core/controllers/access_validators.py

```diff
@@ -305,3 +306,41 @@ def get(self, author_username: str) -> None:
             raise self.PageNotFoundException(
                 'User with given username is not a blog post author.'
             )
+
+
+class TopicEditorPageAccessValidationHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
+    """Validates access to topic editor page."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    URL_PATH_ARGS_SCHEMAS = {
+        'topic_id': {
+            'schema': {
+                'type': 'basestring',
+                'validators': [{
+                    'id': 'is_regex_matched',
+                    'regex_pattern': constants.ENTITY_ID_REGEX
+                }]
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {'GET': {}}
+
+    @acl_decorators.can_view_any_topic_editor
+    def get(self, topic_id: str) -> None:
+        """Displays the topic editor page.
+
+        Args:
+            topic_id: str. The ID of the topic.
+
+        Raises:
+            Exception. The topic with the given id doesn't exist.
+        """
+        topic = topic_fetchers.get_topic_by_id(topic_id, strict=False)
+
+        if topic is None:
+            raise self.PageNotFoundException(
+                Exception('The topic with the given id doesn\'t exist.'))
+        pass
```

Comment:
why do we need pass here?

Reply:
because we removed ` self.render_template('topic-editor-page.mainpage.html')` from previous class "TopicEditorPage" at topic_editor.py

Reply:
So you can just remove that line! (See other handle for reference)

Reply:
Done